addons/spreadsheet_account/models/account.py

```diff
@@ -79,19 +72,182 @@ def _build_spreadsheet_formula_domain(self, formula_params):
+            if end[1] == 'end':
+                depth -= all_boundaries_counter[end]
+
+            period_start = start[0] if start[1] == 'begin' else start[0] + relativedelta(days=1)
+            period_end = end[0] if end[1] == 'end' else end[0] + relativedelta(days=-1)
+
+            # The if condition deals with the issue that occurs when we have:
+            # cell 1 - 01/01/2020 -> 31/12/2020
+            # cell 2 - 01/03/2020 -> 31/03/2020
+            # cell 3 - 01/04/2020 -> 30/04/2020
+            # In such scenario, we actually compute the time period between cell 2 and cell 3 because
+            # we are are in a broader time period due to cell 1. This leads to the following unwanted
+            # time period to be added to the timeline: (01/04/2020, 31/03/2020)
+            if period_start <= period_end:
+                timeline.append((period_start, period_end))
+
+        # Compute in which time period(s) the cell is part of
+        all_starts = [period[0] for period in timeline]
+        all_ends = [period[1] for period in timeline]
+        for args in args_list:
+            start = args['date_from_boundary']
+            end = args['date_to_boundary']
+
+            start_index = all_starts.index(start)
+            end_index = all_ends.index(end)
+
+            args['date_periods'] = timeline[start_index:end_index + 1]
+
+        return [(date(year=1900, month=1, day=1), timeline[0][0] + relativedelta(days=-1))] + timeline
+
+    def _get_all_accounts(self, args_list, default_accounts=False):
+        company_to_codes = defaultdict(set)
+        for args in args_list:
+            codes = {code for code in args["codes"] if code}
+            company_to_codes[args['company_id'] or self.env.company.id].update(codes)
+
+        default_domain = [('account_type', 'in', ['liability_payable', 'asset_receivable'])] if default_accounts else expression.FALSE_DOMAIN
+
+        all_accounts = dict()
+        for company_id, codes in company_to_codes.items():
+            code_domain = expression.OR(
+                [('code', '=like', f'{code}%')]
+                for code in codes
+            )
+            domain = expression.AND([
+                expression.OR([code_domain, default_domain]),
+                self.env['account.account']._check_company_domain(company_id),
+            ])
+            accounts = self.env['account.account'].with_company(company_id).search(domain)
+            all_accounts[company_id] = accounts
+
+        return all_accounts
+
+    def _get_all_lines(self, args_list, fields, timeline, accounts):
+        if not args_list or not fields:
+            return {}
+
+        all_lines = {}
+        aggregates = [field + ':sum' for field in fields]
+
+        include_initial_balance_account_ids = []
+        for company_id, acc in accounts.items():
+            include_initial_balance_account_ids += acc.with_company(company_id).filtered('include_initial_balance').ids
```

Comment:
Why the `with_company`?
```suggestion
            include_initial_balance_account_ids += acc.filtered('include_initial_balance').ids
```